### DISCUSSION TITLE
New bug that never happen in previous version

### DISCUSSION THREAD

**Discussion by XJTLUmedia** [pallets/flask#4803] [Q&A] [answered]
**what the bug is:**
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'index'. Did you mean 'profile' instead?

**Describe how to replicate the bug.**

Running code from below, which is a guide showing in flask document quickstart

```
`from flask import Flask,url_for

app = Flask(__name__)

@app.route("/")
def hello_world():
    return "<p>Hello, World!</p>"

from markupsafe import escape

@app.route('/user/<username>')
def show_user_profile(username):
    # show the user profile for that user
    return f'User {escape(username)}'

@app.route('/post/<int:post_id>')
def show_post(post_id):
    # show the post with the given id, the id is an integer
    return f'Post {post_id}'

@app.route('/user/<username>')
def profile(username):
    return f'{username}\'s profile'

with app.test_request_context():
    print(url_for('index'))
    print(url_for('login'))
    print(url_for('login', next='/'))
    print(url_for('profile', username='John Doe'))`
```

**Describe the expected behavior that should have happened but didn't.**

No errors for these code for previous version of flask. So there should be no problem for this version also

**Environment:**

- Python version: 3.10.6
- Flask version: 2.2.2

**Comment by tachyondecay** [comment, marked as answer]
Your code is missing an `index()` view function. The quickstart’s version is:

~~~py
@app.route('/')
def index():
    return 'index'
~~~

Without the above code, calling `url_for('index')` naturally results in a `BuildError`.
